### ISSUE TITLE
[Support]: Sometimes video is not recorded

### ISSUE THREAD

**Issue by LucaDiba** [blakeblackshear/frigate#2784]
### Describe the problem you are having

After describing my problem in https://github.com/blakeblackshear/frigate/issues/803#issuecomment-1019332604, since the issue is still present even in the last beta, I'm opening a new issue about it.

I'm using a Raspberry Pi 4 8GB. Everything runs on the SD card, while the recordings are stored on an external HD.
I'm recording 24/7 in SQ and on motion only in HD. I noticed that sometimes parts of the 24/7 recording are missing. Sometimes I can find the missing parts in the HD recording, sometimes the missing part is not recorded at all.

I wanted to try to record both SQ and HD 24/7 (so disabling the motion detection at all, at least until I'll be able to have a Coral) to see whether the problem is motion detection. Could this be helpful?

Also, I noticed that sometimes the live video on the Frigate web interface lags a lot, even if, when it lags, everything is stored correctly.

EDIT: I just found out some errors in the log file that you can find below.

### Version

0.10.0-AD4929C

### Frigate config file

```yaml
mqtt:
  host: ***
  user: ***
  password: ***
  
birdseye:
  enabled: False

database:
  path: /db/frigate.db

cameras:

  strada:
    ffmpeg:
      inputs:
        - path: rtsp://***:***@***:554/
          roles:
            - record
        - path: rtsp://***:***@***:554/h264Preview_01_sub
          roles:
            - rtmp
            - detect
    
    detect:
      enabled: False

    record:
      enabled: True
      retain:
        days: ***
        mode: motion

  strada_24_7_record:
    ffmpeg:
      inputs:
        - path: rtsp://***:***@***:554/h264Preview_01_sub
          roles:
            - record
    record:
      enabled: True
      retain:
        days: ***
        mode: all
    
    detect:
      enabled: False

detectors:
  cpu1:
    type: cpu
  cpu2:
    type: cpu
```

### Relevant log output

```shell
[2022-02-10 18:00:07] frigate.video                  ERROR   : strada_24_7_record: Unable to read frames from ffmpeg process.
[2022-02-10 18:00:07] frigate.video                  ERROR   : strada_24_7_record: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:00:16] watchdog.strada_24_7_record    ERROR   : Ffmpeg process crashed unexpectedly for strada_24_7_record.
[2022-02-10 18:00:16] watchdog.strada_24_7_record    ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:00:16] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x557b510a40] Failed to update header with correct duration.
[2022-02-10 18:00:16] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x557b510a40] Failed to update header with correct filesize.
[2022-02-10 18:00:32] frigate.video                  ERROR   : strada_24_7_record: Unable to read frames from ffmpeg process.
[2022-02-10 18:00:32] frigate.video                  ERROR   : strada_24_7_record: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:00:36] watchdog.strada_24_7_record    ERROR   : Ffmpeg process crashed unexpectedly for strada_24_7_record.
[2022-02-10 18:00:36] watchdog.strada_24_7_record    ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:00:36] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x5599486a40] Failed to update header with correct duration.
[2022-02-10 18:00:36] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x5599486a40] Failed to update header with correct filesize.
[2022-02-10 18:00:47] frigate.video                  ERROR   : strada: Unable to read frames from ffmpeg process.
[2022-02-10 18:00:47] frigate.video                  ERROR   : strada: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:00:56] watchdog.strada                ERROR   : Ffmpeg process crashed unexpectedly for strada.
[2022-02-10 18:00:56] watchdog.strada                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:00:56] ffmpeg.strada.detect           ERROR   : [flv @ 0x55918699a0] Failed to update header with correct duration.
[2022-02-10 18:00:56] ffmpeg.strada.detect           ERROR   : [flv @ 0x55918699a0] Failed to update header with correct filesize.
[2022-02-10 18:01:27] frigate.video                  ERROR   : strada_24_7_record: Unable to read frames from ffmpeg process.
[2022-02-10 18:01:27] frigate.video                  ERROR   : strada_24_7_record: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:01:36] watchdog.strada_24_7_record    ERROR   : Ffmpeg process crashed unexpectedly for strada_24_7_record.
[2022-02-10 18:01:36] watchdog.strada_24_7_record    ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:01:36] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x556f0bda10] Failed to update header with correct duration.
[2022-02-10 18:01:36] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x556f0bda10] Failed to update header with correct filesize.
[2022-02-10 18:02:07] frigate.video                  ERROR   : strada: Unable to read frames from ffmpeg process.
[2022-02-10 18:02:07] frigate.video                  ERROR   : strada: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:02:16] watchdog.strada                ERROR   : Ffmpeg process crashed unexpectedly for strada.
[2022-02-10 18:02:16] watchdog.strada                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:02:16] ffmpeg.strada.detect           ERROR   : [flv @ 0x55aaa419a0] Failed to update header with correct duration.
[2022-02-10 18:02:16] ffmpeg.strada.detect           ERROR   : [flv @ 0x55aaa419a0] Failed to update header with correct filesize.
[2022-02-10 18:02:47] frigate.video                  ERROR   : strada_24_7_record: Unable to read frames from ffmpeg process.
[2022-02-10 18:02:47] frigate.video                  ERROR   : strada_24_7_record: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:02:56] watchdog.strada_24_7_record    ERROR   : Ffmpeg process crashed unexpectedly for strada_24_7_record.
[2022-02-10 18:02:56] watchdog.strada_24_7_record    ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:02:56] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x55b718fe80] Failed to update header with correct duration.
[2022-02-10 18:02:56] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x55b718fe80] Failed to update header with correct filesize.
[2022-02-10 18:03:27] frigate.video                  ERROR   : strada: Unable to read frames from ffmpeg process.
[2022-02-10 18:03:27] frigate.video                  ERROR   : strada: Unable to read frames from ffmpeg process.
[2022-02-10 18:03:27] frigate.video                  ERROR   : strada: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:03:36] watchdog.strada                ERROR   : Ffmpeg process crashed unexpectedly for strada.
[2022-02-10 18:03:36] watchdog.strada                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:03:36] ffmpeg.strada.detect           ERROR   : [flv @ 0x55ab2609a0] Failed to update header with correct duration.
[2022-02-10 18:03:36] ffmpeg.strada.detect           ERROR   : [flv @ 0x55ab2609a0] Failed to update header with correct filesize.
[2022-02-10 18:04:07] frigate.video                  ERROR   : strada_24_7_record: Unable to read frames from ffmpeg process.
[2022-02-10 18:04:07] frigate.video                  ERROR   : strada_24_7_record: Unable to read frames from ffmpeg process.
[2022-02-10 18:04:07] frigate.video                  ERROR   : strada_24_7_record: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:04:16] watchdog.strada_24_7_record    ERROR   : Ffmpeg process crashed unexpectedly for strada_24_7_record.
[2022-02-10 18:04:16] watchdog.strada_24_7_record    ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:04:16] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x559ecbfe80] Failed to update header with correct duration.
[2022-02-10 18:04:16] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x559ecbfe80] Failed to update header with correct filesize.
[2022-02-10 18:04:47] frigate.video                  ERROR   : strada: Unable to read frames from ffmpeg process.
[2022-02-10 18:04:47] frigate.video                  ERROR   : strada: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:04:56] watchdog.strada                ERROR   : Ffmpeg process crashed unexpectedly for strada.
[2022-02-10 18:04:56] watchdog.strada                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:04:56] ffmpeg.strada.detect           ERROR   : [flv @ 0x5566fe79a0] Failed to update header with correct duration.
[2022-02-10 18:04:56] ffmpeg.strada.detect           ERROR   : [flv @ 0x5566fe79a0] Failed to update header with correct filesize.
[2022-02-10 18:05:27] frigate.video                  ERROR   : strada_24_7_record: Unable to read frames from ffmpeg process.
[2022-02-10 18:05:27] frigate.video                  ERROR   : strada_24_7_record: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:05:36] watchdog.strada_24_7_record    ERROR   : Ffmpeg process crashed unexpectedly for strada_24_7_record.
[2022-02-10 18:05:36] watchdog.strada_24_7_record    ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:05:36] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x55b8e56fc0] Failed to update header with correct duration.
[2022-02-10 18:05:36] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x55b8e56fc0] Failed to update header with correct filesize.
[2022-02-10 18:06:07] frigate.video                  ERROR   : strada: Unable to read frames from ffmpeg process.
[2022-02-10 18:06:07] frigate.video                  ERROR   : strada: Unable to read frames from ffmpeg process.
[2022-02-10 18:06:07] frigate.video                  ERROR   : strada: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:06:16] watchdog.strada                ERROR   : Ffmpeg process crashed unexpectedly for strada.
[2022-02-10 18:06:16] watchdog.strada                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:06:16] ffmpeg.strada.detect           ERROR   : [h264 @ 0x557a1cc910] error while decoding MB 12 22, bytestream -5
[2022-02-10 18:06:16] ffmpeg.strada.detect           ERROR   : rtsp://***:***@***:554/h264Preview_01_sub: corrupt decoded frame in stream 0
[2022-02-10 18:06:16] ffmpeg.strada.detect           ERROR   : [flv @ 0x557a1889a0] Failed to update header with correct duration.
[2022-02-10 18:06:16] ffmpeg.strada.detect           ERROR   : [flv @ 0x557a1889a0] Failed to update header with correct filesize.
[2022-02-10 18:06:27] frigate.video                  ERROR   : strada_24_7_record: Unable to read frames from ffmpeg process.
[2022-02-10 18:06:27] frigate.video                  ERROR   : strada_24_7_record: Unable to read frames from ffmpeg process.
[2022-02-10 18:06:27] frigate.video                  ERROR   : strada_24_7_record: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:06:36] watchdog.strada_24_7_record    ERROR   : Ffmpeg process crashed unexpectedly for strada_24_7_record.
[2022-02-10 18:06:36] watchdog.strada_24_7_record    ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:06:36] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x55b931bf60] Failed to update header with correct duration.
[2022-02-10 18:06:36] ffmpeg.strada_24_7_record.detect ERROR   : [flv @ 0x55b931bf60] Failed to update header with correct filesize.
[2022-02-10 18:07:07] frigate.video                  ERROR   : strada: Unable to read frames from ffmpeg process.
[2022-02-10 18:07:07] frigate.video                  ERROR   : strada: ffmpeg process is not running. exiting capture thread...
[2022-02-10 18:07:16] watchdog.strada                ERROR   : Ffmpeg process crashed unexpectedly for strada.
[2022-02-10 18:07:16] watchdog.strada                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
[2022-02-10 18:07:16] ffmpeg.strada.detect           ERROR   : [flv @ 0x5591e9f9a0] Failed to update header with correct duration.
[2022-02-10 18:07:16] ffmpeg.strada.detect           ERROR   : [flv @ 0x5591e9f9a0] Failed to update header with correct filesize.
```

### FFprobe output from your camera

```shell
$ ffprobe rtsp://***:***@***:554
ffprobe version 4.2.4-1ubuntu0.1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-10ubuntu2)
  configuration: --prefix=/usr --extra-version=1ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
[rtsp @ 0x55a279069a00] UDP timeout, retrying with TCP
Input #0, rtsp, from 'rtsp://***:***@***:554':
  Metadata:
    title           : Session streamed by "preview"
    comment         :
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0: Video: hevc (Main), yuv420p(tv), 3840x2160, 24.92 tbr, 90k tbn, 90k tbc
    Stream #0:1: Audio: aac (LC), 16000 Hz, mono, fltp
```

### Frigate stats

```json
{
   "detection_fps":0.0,
   "detectors":{
      "cpu1":{
         "detection_start":0.0,
         "inference_speed":10.0,
         "pid":216
      },
      "cpu2":{
         "detection_start":0.0,
         "inference_speed":10.0,
         "pid":218
      }
   },
   "service":{
      "storage":{
         "/dev/shm":{
            "free":63.9,
            "mount_type":"tmpfs",
            "total":67.1,
            "used":3.2
         },
         "/media/frigate/clips":{
            "free":1613839.7,
            "mount_type":"ext4",
            "total":1968873.2,
            "used":254996.8
         },
         "/media/frigate/recordings":{
            "free":1613839.7,
            "mount_type":"ext4",
            "total":1968873.2,
            "used":254996.8
         },
         "/tmp/cache":{
            "free":998.7,
            "mount_type":"tmpfs",
            "total":1000.0,
            "used":1.3
         }
      },
      "temperatures":{
         
      },
      "uptime":47483,
      "version":"0.10.0-ad4929c"
   },
   "strada":{
      "camera_fps":0.1,
      "capture_pid":227,
      "detection_fps":0.0,
      "pid":222,
      "process_fps":0.1,
      "skipped_fps":0.0
   },
   "strada_24_7_record":{
      "camera_fps":5.0,
      "capture_pid":229,
      "detection_fps":0.0,
      "pid":225,
      "process_fps":5.0,
      "skipped_fps":0.0
   }
}
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

Reolink 810A

**Comment by blakeblackshear** [comment]
It looks like the connection to your camera is unstable. Any time a disconnect happens, you will lose that 10s segment. This is typical for Reolink cameras. Any reason you aren't using the recommended settings for Reolink from the documentation?

**Comment by LucaDiba** [comment]
If I remember correctly, I haven't used that configuration since it wasn't working for my camera. Since the documentation says that it's for RLC 410 and 520 only, I thought that it wasn't for mine.

I just tried again and this is what I found out:
- the `ext` and `sub` channels URLs work properly (`http://[ip]/flv?port=1935&app=bcs&stream=channel0_[ext/sub].bcs&user=[user]&password=[password]`)
- the `main` channel URL doesn't work

Since I was trying the same configuration and the example in the documentation was using `main` channel, it didn't work.

Regarding the `main` channel, [it seems that there is no HTTP URL](https://forums.zoneminder.com/viewtopic.php?p=120273#p120273), since `main` uses H.265, so only the RTMP will work.

Right now I set up the HTTP for the 24/7 recording and the RTMP for the clear (`main`) channel.
Tomorrow I'll try to check better if I find something else; I'll also check whether using the HTTP solved the problem, at least for the `ext` channel (right now it seems waaaaay faster and more stable 😄 ). I'll keep you updated.

Thanks a lot for your help!

PS: in the documentation, at the page `/configuration/camera_specific/`, the row `hwaccel_args:` on the Reolink configuration should be updated to `hwaccel_args: []` otherwise the following error will be raised:
```
cameras -> strada_24_7_record -> ffmpeg -> hwaccel_args
  none is not an allowed value (type=type_error.none.not_allowed)
```

PPS:
I also had to remove the two following lines:
```
- -rw_timeout
- "5000000"
```
otherwise it kept giving the following error in loop:
```
[2022-02-11 02:33:49] ffmpeg.strada.record           ERROR   : Option rw_timeout not found.
[2022-02-11 02:33:49] watchdog.strada                INFO    : Terminating the existing ffmpeg process...
[2022-02-11 02:33:49] watchdog.strada                INFO    : Waiting for ffmpeg to exit gracefully...
```
Since I now removed this part, might it be a problem or it doesn't matter?
